hibernate-core/src/main/java/org/hibernate/query/sqm/tuple/internal/AnonymousTupleType.java

```diff
@@ -72,9 +70,13 @@ public AnonymousTupleType(SqmSelectableNode<?>[] components) {
 		//noinspection unchecked
 		javaTypeDescriptor = (JavaType<T>) new ObjectArrayJavaType( getTypeDescriptors( components ) );
 		componentIndexMap = linkedMapOfSize( components.length );
+		final String[] aliases = extractAliases( selectQuery );
 		for ( int i = 0; i < components.length; i++ ) {
 			final SqmSelectableNode<?> component = components[i];
-			final String alias = component.getAlias();
+			String alias = aliases[i];
+			if ( alias == null ) {
+				alias = component.getAlias();
+			}
```

Comment:
Still not sure why we need this check. Aren't you already falling back to the selectable node's alias in `extractAliases`?